packages/adaptive_scaffold/lib/src/slot_layout_config.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+/// A Widget that takes a child and a Function that returns an animated Widget
+/// then displays the child with this animation.
+///
+/// Most commonly used in [LayoutSlot] however it would be functional outside of
+/// this Widget as well.
```

Comment:
Why do we want to keep it this way? I'd prefer making it impossible to use outside of `LayoutSlot` to prevent misuse. The widget that `LayoutSlot` displays based on `SlotLayoutConfig` should be private.

Reply:
I've changed SlotLayoutConfig to an abstract class and added a function to SlotLayout called SlotLayout.from (naming can be changed). SlotLayout.from just redirects the values to a private implementation of the abstract SlotLayoutConfig. I did it this way because the type SlotLayoutConfig needs to be public when describing what is inside the map but if theres some way around this then please let me know. Thanks.